internal/pkg/config/config.go

```diff
@@ -147,8 +147,9 @@ type fieldTagMatcher struct {
 
 // this type uses the default unmarshaller and mirrors configuration key-value pairs
 type rawFieldTagMatcher struct {
```

Comment:
Wont this allow to be parsed?
FieldTags:
  - Key: datapolicy
    Val: cert
    Value: password

Reply:
That is correct. In that case we should probably do like we are doing with e.g. `Package` and `PackageRE`, and only allow one or the other to be set. Although, to be honest I see little value in having both `Val` and `Value`, so I think we may actually want to keep only one of them. WDYT?

Reply:
Yeah lets just keep `Value` and treat setting `Val` as an error. Being strict while parsing the config is totally fine IMO.

Reply:
@PurelyApplied, @benhxy, thoughts on this?

Reply:
Oh, I had a WIP branch that resolved some of this with `RawMessage` and `switch` statements.  \*dig dig dig\*

#271 opened as draft.  I hadn't added tests yet, but it hits some of the concerns you've voiced.  There's also the issue where, if we're case insensitive, you could theoretically add all of `key`, `Key`, `KEY`, etc and have collisions that way while still being valid YAML.  #271 pokes at the other matchers, too, since I wasn't a huge fan of that Valid Keys slice.

> That is correct. In that case we should probably do like we are doing with e.g. Package and PackageRE, and only allow one or the other to be set. Although, in my opinion there is little value in having both Val and Value, so I think we may actually want to keep only one of them. WDYT?

I'd be +1 to only retaining `Value`.  I think we'd kept `Val` because we didn't want to rewrite existing test configs.  But we're v0.x.  We can make a config-breaking change here.